Modules/System/Barcode/src/Barcode Provider 2D/Image/BarcodeImageEncoder2D.Interface.al

```diff
@@ -0,0 +1,18 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary> 
+/// Exposes common interface for 2D barcode image encoder.
+/// </summary>
+interface "Barcode Image Encoder 2D"
+{
+    /// <summary> 
+    /// Encodes a input text to a barcode image.
+    /// </summary>
+    /// <param name="InputText">The text to encode.</param>
+    /// <param name="BarcodeEncodeSettings2D">The settings to use when encoding the text.</param>
+    /// <returns>The encoded barcode.</returns>
+    procedure EncodeImage(InputText: Text; BarcodeEncodeSettings2D: Record "Barcode Encode Settings 2D"): Codeunit "Temp Blob";
```

Comment:
Would it not make sense to have an overload without the settings parameter table? Does one always have to explicitly define the options, or are there default settings, which often would do the trick?

Reply:
I guess you copied the implementation we've got in 1D barcodes? In that case, we're probably ok.

Reply:
There is already an overload in interface "Barcode Image Provider 2D".
Normally I wouldn't call the interface "Barcode Image Encoder 2D" directly, , would I?

Reply:
Nope, we're good here.